src/main/java/dev/openfeature/javasdk/EvaluationContext.java

```diff
@@ -133,11 +133,11 @@ public static EvaluationContext merge(EvaluationContext ctx1, EvaluationContext
         ec.attributes.putAll(ctx1.attributes);
         ec.attributes.putAll(ctx2.attributes);
 
-        if (ctx1.getTargetingKey() != null) {
+        if (ctx1.getTargetingKey() != null && !ctx1.getTargetingKey().equals("")) {
```

Comment:
Oof! Can you write a regression test for this?

Reply:
Apache StringUtils has a bunch of convenience helper methods for this, eg https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotEmpty-java.lang.CharSequence-

For larger projects, I always tend to include this because it's simpler and less prone to errors... but do we want to introduce the extra dependencies in the SDK, making it bigger?

Reply:
(there's also `isNotBlank` which will treat all-whitespace strings like empty strings)

Reply:
Added a test. Also added a trim to the check to handle whitespace, the java sdk is set to Java 8 and isNotBlank is part of 11 so didn't use that.